src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -44,57 +48,182 @@ export class PermissionsTableComponent implements OnInit {
-    this.otherEntities = this.getOtherEntities(this.entityType);
+
+    this.rolesCheckbox = {};
+    this.roles.forEach(r => {
+      this.rolesCheckbox[r] = false;
+    });
   }
 
-  private getEntities(entityType: string): Entity[] {
-    switch (entityType) {
-      case 'users':
-        // TODO get from rolesService instead
-        return [
-          {name: 'marvec', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'livthomas', roles: ['read', 'share', 'write']}
-        ];
-      case 'groups':
-        // TODO get from rolesService instead
-        return [
-          {name: 'managers', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'employees', roles: ['read', 'share']},
-          {name: 'contractors', roles: ['read']}
-        ];
-      default:
-        throw Error('unknown entity type');
+  private getEntities(entityType: string, permissions: Permissions): Permission[] {
+    let entities = permissions[entityType];
+    if (entities) {
+      return entities;
+    } else {
+      throw Error('unknown entity type');
+    }
+  }
+
+  private setPermissionsForResource(resourceType: string) {
+    let parent = this;
+    let getPermissionsFunctions = {
+      'organization': function() {return parent.organizationService.getPermissions();} ,
+      'project': function() {return parent.projectService.getPermissions();},
+      'collection': null, // TODO
+      'view': null // TODO
+    };
+    let getPermissions = getPermissionsFunctions[resourceType];
+    if (getPermissions) {
+      getPermissions().subscribe((permissions: Permissions) => {
+          this.entities = this.getEntities(this.entityType, permissions);
+          this.otherEntities = this.getOtherEntities(this.entityType);
+        });
+    } else {
+      throw Error('unknown resorce type');
     }
   }
 
   private getOtherEntities(entityType: string): string[] {
     switch (entityType) {
       case 'users':
         // TODO get from usersService instead
-        return ['alicak', 'kubedo', 'jkotrady', 'kulexpipiens'];
+        let userEntities = ['alicak', 'kubedo', 'jkotrady', 'kulexpipiens'];
+        for (let entity of this.entities) {
+          let index = userEntities.indexOf(entity['name']);
+          if (index >= 0) {
+            userEntities.splice(index, 1);
+          }
+        }
+        return userEntities;
       case 'groups':
         // TODO get from groupsService instead
-        return ['directors', 'customers'];
+        let groupEntities = ['directors', 'customers'];
+        for (let entity of this.entities) {
+          let index = groupEntities.indexOf(entity['name']);
+          if (index >= 0) {
+            groupEntities.splice(index, 1);
+          }
+        }
+        return groupEntities;
       default:
         throw Error('unknown entity type');
     }
   }
 
-  public onAdd() {
-    // TODO call remote service
+  public onAdd(selectedName: string) {
```

Comment:
You can split the function into smaller parts (`getResource` and `updatePrmissions`).